orchestrator/src/main/java/eu/arrowhead/core/orchestrator/database/service/OrchestratorStoreFlexibleDBService.java

```diff
@@ -0,0 +1,188 @@
+	}
+	
+	//-------------------------------------------------------------------------------------------------
+	public OrchestratorStoreFlexible createOrchestratorStoreFlexible(final String consumerSystemName, final String providerSystemName, final String consumerSystemMetadata, final String providerSystemMetadata,
+			 														 final String serviceMetadata, final String serviceInterfaceName, final String serviceDefinitionName, final Integer priority) { //TODO junit
+		logger.debug("createOrchestratorStoreFlexible started...");
+		final OrchestratorStoreFlexible candidate = new OrchestratorStoreFlexible(consumerSystemName, providerSystemName, consumerSystemMetadata, providerSystemMetadata, serviceMetadata,
+																				  serviceInterfaceName, serviceDefinitionName, priority);
+		validateEntity(candidate);
+		return saveEntityList(List.of(candidate)).get(0);
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	public List<OrchestratorStoreFlexible> createOrchestratorStoreFlexible(final List<OrchestratorStoreFlexible> candidates) { //TODO junit
+		logger.debug("createOrchestratorStoreFlexible started...");
+		Assert.notNull(candidates, "OrchestratorStoreFlexible candidate list is null");
+		
+		for (final OrchestratorStoreFlexible candidate : candidates) {
+			validateEntity(candidate);
+		}
+		
+		return saveEntityList(candidates);
+	}
+	
+	//-------------------------------------------------------------------------------------------------
+	@Transactional(rollbackFor = ArrowheadException.class)
+	public void deleteOrchestratorStoreFlexibleById(final long id) { //TODO junit
+		logger.debug("deleteOrchestratorStoreFlexibleById started...");
+		
+		try {
+			if (orchestratorStoreFlexibleRepository.existsById(id)) {
+				orchestratorStoreFlexibleRepository.deleteById(id);
+			}
+		} catch (final Exception ex) {
+			logger.debug(ex.getMessage(), ex);
+			throw new ArrowheadException(CoreCommonConstants.DATABASE_OPERATION_EXCEPTION_MSG);
+		}
+	}
+	
+	//-------------------------------------------------------------------------------------------------
+	@Transactional(rollbackFor = ArrowheadException.class)
+	public void deleteAllOrchestratorStoreFlexible() { //TODO junit
+		logger.debug("deleteAllOrchestratorStoreFlexible started...");
+		
+		try {
+			orchestratorStoreFlexibleRepository.deleteAll();
+		} catch (final Exception ex) {
+			logger.debug(ex.getMessage(), ex);
+			throw new ArrowheadException(CoreCommonConstants.DATABASE_OPERATION_EXCEPTION_MSG);
+		}
+	}
+	
+	//=================================================================================================
+	// assistant methods
+	
+	//-------------------------------------------------------------------------------------------------
+	private void validateEntity(final OrchestratorStoreFlexible entity) {
```

Comment:
see suggested checks in the controller methods
somewhere you should normalize the names (like toLowercase().trim() in case of names and service def, and uppercase in case of interface)
check for non-positive priority (you should check this in the controller methods as well)